ZitiNativeApiForDotnetCore/library/ziti4dotnet.c

```diff
@@ -86,4 +88,67 @@ int ziti_context_event_status(const ziti_event_t* e) {
     else {
         return 0;
     }
-}
\ No newline at end of file
+}
+
+typedef int (*printer)(void* arg, const char* fmt, ...);
+
+static int ziti_dump_to_log_op(void* stringsBuilder, const char* fmt, ...) {
+    static char line[4096];
+
+    va_list vargs;
+    va_start(vargs, fmt);
+    vsnprintf(line, sizeof(line), fmt, vargs);
+    va_end(vargs);
+
+    if (strlen(stringsBuilder) + strlen(line) > MAXBUFFERLEN) {
+        return -1;
+    }
+    // write/append to the buffer
+    strncat(stringsBuilder, line, sizeof(line));
+    return 0;
+}
+
+static int ziti_dump_to_file_op(void* fp, const char* fmt, ...) {
+    static char line[4096];
+
+    va_list vargs;
+    va_start(vargs, fmt);
+    // write/append to file
+    vfprintf(fp, fmt, vargs);
+    va_end(vargs);
+
+    return 0;
+}
+
+void z4d_ziti_dump_log(ziti_context ztx) {
+    char* buffer;
+    buffer = malloc(MAXBUFFERLEN * sizeof(char));
+    buffer[0] = 0;
+    ziti_dump(ztx, (printer)ziti_dump_to_log_op, buffer);
+    printf("ziti dump to log %s", buffer);

```

Comment:
we don't need this printf do we???

Reply:
ZITI_LOG is not working in the ziti4dotnet.c, so I used printf statements. I may need some more time to check why ZITI_LOG is not working in native library